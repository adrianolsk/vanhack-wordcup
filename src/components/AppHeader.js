import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const StyledAppHeader = styled.div`
  background-color: #0b5f91;
  width: 100%;
  h1 {
    margin: 0;
    padding: 20px;
    color: #fff;
  }
`;

const AppHeader = ({ title, children }) => (
  <StyledAppHeader>
    <h1>{title}</h1>
    {children}
  </StyledAppHeader>
);

AppHeader.propTypes = {
  title: PropTypes.string.isRequired
};
export default AppHeader;
