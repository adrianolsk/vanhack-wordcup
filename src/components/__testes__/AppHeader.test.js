import React from "react";
import AppHeader from "../AppHeader";

import { render, Simulate, wait } from "react-testing-library";
// import "dom-testing-library/extend-expect";

describe(" <AppHeader/>", () => {
  const title = "Test";

  const onChangeSpy = jest.fn();
  const { getByText, getByTestId, container, queryByText } = render(
    <AppHeader title={title} />
  );

  it("renders", () => {
    const titleOfTheHeader = queryByText("Test");
    expect(titleOfTheHeader).toBeTruthy(); // it exist
  });

  it("match snapshot", () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});
