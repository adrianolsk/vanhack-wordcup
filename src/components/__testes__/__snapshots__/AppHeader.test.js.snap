// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` <AppHeader/> match snapshot 1`] = `
<div
  class="sc-bdVaJa mjvwZ"
>
  <h1>
    Test
  </h1>
</div>
`;
